pkg/crypto/rings/client.go

```diff
@@ -0,0 +1,209 @@
+package rings
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/depinject"
+	ring_secp256k1 "github.com/athanorlabs/go-dleq/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/noot/ring-go"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	"github.com/pokt-network/poktroll/pkg/crypto"
+	pubkeyclient "github.com/pokt-network/poktroll/pkg/crypto/pubkey_client"
+	"github.com/pokt-network/poktroll/pkg/polylog"
+	"github.com/pokt-network/poktroll/x/service/types"
+)
+
+var _ crypto.RingClient = (*ringClient)(nil)
+
+// ringClient is an implementation of the RingClient interface that uses the
+// client.ApplicationQueryClient to get applications delegation information and
+// needed to construct the ring for signing relay requests.
+// It also uses the pubkeyclient.PubKeyClient to get the public keys for the
+// addresses in the ring.
```

Comment:
Aren't these two sentences redundant? We need the public keys to construct the ring so I feel like we can just delete it.